dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/utils/HttpUtils.java

```diff
@@ -467,20 +466,90 @@ public static String getProtocolHeader(Message m, String name, String defaultVal
 
     public static String getBaseAddress(Message m) {
         String endpointAddress = getEndpointAddress(m);
-        try {
-            URI uri = new URI(endpointAddress);
-            String path = uri.getRawPath();
-            String scheme = uri.getScheme();
+        //Liberty code change start
+        String[] addr = parse(endpointAddress, false);
+        if (addr != null) {
+            String scheme = addr[0];
+            String path = addr[1];
             if (scheme != null && !scheme.startsWith(HttpUtils.HTTP_SCHEME)
-                && HttpUtils.isHttpRequest(m)) {
+                    && HttpUtils.isHttpRequest(m)) {
                 path = HttpUtils.toAbsoluteUri(path, m).getRawPath();
             }
             return (path == null || path.length() == 0) ? "/" : path;
-        } catch (URISyntaxException ex) {
+        } else {
             return endpointAddress == null ? "/" : endpointAddress;
         }
     }
 
+    public static String[] parse(String addr, boolean parseAuthority) {
```

Comment:
I kinda want this method to have something beside just the word parse.  Maybe parseAddress or parseURI, or parseEndpointAddress or something.